types/base.d.ts

```diff
@@ -5,6 +5,7 @@ export declare class API {
 
     constructor(options?: Record<string, unknown>);
     trigger(topic: string, message: Message, options?: Record<string, unknown>): Promise<void>;
+    bind(topic: string, options?: Function | object): Promise<void>

```

Comment:
Fully type `options` by including the full set of option objects and functions supported.

Reply:
Otherwise, this typing doesn't help much because as a user I might by passing on options or functions that don't make any sense.